docs/quickstart/1-install-requirements.md

```diff
@@ -15,6 +15,18 @@ Follow the instructions based on your OS:
 
 ## Docker Compose
 
-**source{d} CE** is deployed as Docker containers, using [Docker Compose](https://docs.docker.com/compose), but it is not required to have a local installation of Docker Compose; if it is not found it will be deployed inside a container.
+**source{d} CE** is deployed as Docker containers, using [Docker Compose](https://docs.docker.com/compose), but it is not required to have a local installation of Docker Compose; if it is not found it will be downloaded from docker sources, and deployed inside a container.
 
-If you prefer a local installation of Docker Compose, you can follow the [Docker Compose install guide](https://docs.docker.com/compose/install)
+If you prefer a local installation of Docker Compose, or you have no access to internet to download it, you can follow the [Docker Compose install guide](https://docs.docker.com/compose/install)
+
+## Internet Connection
```

Comment:
Most people will not need this completely offline use case. And for people behind a firewall, I assume most of the times they will make it work configuring their firewall rules.
I think this section is a bit out of place in the quickstart, and it can be confusing to explain a use case that is not the common one, or supported.

What do you think about moving this to the FAQ? This way we have a place to link if somebody asks, but it's not distracting to first time users.

Reply:
I'm not strongly against moving this into our FAQ, but I really see the connection as a requirement to be made explicit when the user is installing source{d}.

We both agree that the intended use of source{d} requires Internet, but in the last few weeks, we found a couple of people having this very same problem https://github.com/src-d/sourced-ce/issues/206 https://github.com/src-d/sourced-ce/issues/241, what I think it's legit because of the target of this tool (being used by engineers to try its features, to be applied in big corps, where there can be firewalls or other net limitations).

On the other side, this doc does not explain how to use the tool without net connectivity, but only list the resources to be requested from the net, that should be satisfied if there is no connectivity available.

From the reader pov, I'd find easy to skip the whole section if they have connectivity, and just read it if they don't (what might be interesting considering what was exposed above).

So, if the goal is to have a shorter quickstart as possible, and you agree on the importance of not considering Internet as a supposed resource, would you find better if we just:
- mention the connectivity requirement in this section, and
- move the extra details about the required resources & what to do if not available to the FAQ.

Reply:
My view in general is that for docs, less is more. In a way I think that documenting too much is similar to feature creep in the code. This section is not bad. It's not misleading or confusing. But if we go this direction, things add up, and then it becomes noise in my opinion. Should we also include in the quickstart that you need to make sure the ports we'll try to use are free? storage requirements? Or taking it to the extreme: should we explain custom deployments, like for example how deploy connecting to a postgresql already deployed in your company network?

For the issues you link:
#206 already contains the error `Get https://.... docker-compose.yml: Forbidden`. The documentation here does not add any extra information that would help to figure out how to configure the firewall.

In #241 the problem was that they could manually provide the file `sourced-ce` needed, but didn't know where to copy it manually. The improved error messages from this PR would be enough to solve their problem.
In this case this section would also not provide a solution, we say you will need to provide the file manually but don't document where exactly. Also, the docs here link to the docker-compose.yml from master, and that's not correct, it can lead to the deployment not working as expected if the user places this file manually in their `~/.sourced` dir.

Reply:
About #206, when they tried to initialize `sourced init local .`
before:
```
Get https://raw.githubusercontent.com/src-d/sourced-ce/v0.14.0-rc.2/docker-compose.yml: Forbidden
```
The user doesn't know why is that file needed, and how to fix it to be used without internet (because it is not said to be a requirement)

now:
```
[...]: error downloading https://raw.githubusercontent.com/src-d/sourced-ce/master/docker-compose.yml to /.sourced/compose-files/master/docker-compose.yml: [...]

The source{d} CE config file could not be downloaded. see docs about internet requirement.
```
The error message now explains which kind of file is failing and points to docs explaining that internet connection is a requirement. 
The error also explains where should be placed the file (which could be used by a ninja engineer to provide that file)

As agreed, the purpose of the docs is not to explain how to run source{d} offline, but at least explain clearly what and why is failing.
What we had before, already let the user know that a file could not be downloaded, but it seems it was not enough, because they felt the necessity of opening an issue, I guess if it was because they didn't assume that source{d} was not meant to be run offline.

About #241 it's kind of the same. It was needed three interactions to realize that the problem was a lack of connectivity, so I'd say that this assumption might be too much for our users.
For sure that if we start having questions/problems about port conflicts, we'll have to document or provide an alternative, but we can manage that problem when it happens.

Reply:
> As agreed, the purpose of the docs is not to explain how to run source{d} offline, but at least explain clearly what and why is failing.

But the docs here don't explain what and why is failing. The error contains more info that the docs: trying to access the internet, the exact URL, the destination file, and the error.
For #206 the docs don't mention firewalls or `Forbidden` errors, or the destination path for the file. Even the URL is wrong, it points to master when it will be `http...<version>/docker-compose.yml`.


> About #241 it's kind of the same. It was needed three interactions to realize that the problem was a lack of connectivity, so I'd say that this assumption might be too much for our users.

Of course they knew they needed connectivity to download a file! The problem was not that, the problem was that they didn't know where to place the file that was provided manually:

> Edit: Also, I have in my notes that he tried fetching run.sh separately, but could not figure out where to put it, so that it would work correctly with the other files.

In this case, the new errors from this PR would solve the problem. But the docs don't mention the URL or destination path for the missing file, so it would not help in this case either.

> For sure that if we start having questions/problems about port conflicts, we'll have to document or provide an alternative, but we can manage that problem when it happens.

I was actually arguing for the exact opposite... to **not** document every possible problem. `sourced init` is already capable of telling the users when the containers cannot be created due to a port conflict.